tests/distributed/test_tp_gptq.py

```diff
@@ -0,0 +1,730 @@
+                    tp_w1_scales
+                )
+                torch.testing.assert_close(
+                    wg1_fused_qzeros[:, w1_out_idx_start // packing : w1_out_idx_end // packing],
+                    tp_w1_qzeros
+                )
+                if config.use_bias:
+                    torch.testing.assert_close(
+                        wg1_fused_bias[w1_out_idx_start : w1_out_idx_end],
+                        tp_w1_bias
+                    )
+                else:
+                    assert (
+                        not hasattr(tp_mod.wg1_fused, "bias")
+                        or tp_mod.wg1_fused.bias is None
+                        or torch.sum(tp_mod.wg1_fused.bias) == 0
+                    )
+                    assert (
+                        wg1_fused_qparam.get("bias", None) is None
+                        or torch.sum(wg1_fused_qparam.get("bias")) == 0
+                    )
+
+                # load pre-TP tensors of w2 linear module
+                w2_qw = qparams["w2.qweight"]
+                w2_scales = qparams["w2.scales"]
+                w2_qzeros = qparams["w2.qzeros"]
+                w2_gidx = qparams["w2.g_idx"]
+                if config.use_bias:
+                    w2_bias = qparams["w2.bias"]
+
+                # assert on w2 linear module
+                w2_mult = config.multiple_of * config.emb_dim // group.size()
+                w2_mult_packed = w2_mult // packing
+                w2_mult_grp = w2_mult // config.group_size
+                torch.testing.assert_close(
+                    w2_qw[w2_mult_packed * rank : w2_mult_packed * (rank + 1)],
+                    tp_mod.w2.qweight,
+                )
+                torch.testing.assert_close(
+                    w2_scales[w2_mult_grp * rank : w2_mult_grp * (rank + 1)],
+                    tp_mod.w2.scales,
+                )
+                torch.testing.assert_close(
+                    w2_qzeros[w2_mult_grp * rank : w2_mult_grp * (rank + 1)],
+                    tp_mod.w2.qzeros,
+                )
+                w2_gidx_unscaled = w2_gidx[w2_mult * rank : w2_mult * (rank + 1)]
+                torch.testing.assert_close(
+                    w2_gidx_unscaled - min(w2_gidx_unscaled),
+                    tp_mod.w2.g_idx,
+                )
+                if config.use_bias:
+                    torch.testing.assert_close(
+                        w2_bias,
+                        tp_mod.w2.bias,
+                    )
+                else:
+                    assert (
+                        not hasattr(tp_mod.w2, "bias")
+                        or tp_mod.w2.bias is None
+                        or torch.sum(tp_mod.w2.bias) == 0
+                    )
+                    assert (
+                        w2_qparam.get("bias", None) is None
+                        or torch.sum(w2_qparam.get("bias")) == 0
+                    )
+
+        # Test world_size < kvheads
```

Comment:
I would parameterize the test with the MockGroup world_size that way we would create 4 tests here and pass the tests that work rather than fail 1 large test